### PR TITLE
fix(query): use custom connector for udf client

### PR DESCRIPTION
**Pull request by everpcpc**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- enable  SO_REUSEADDR for udf client
- use global dns resolver

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16697)

### CODE REVIEW

File: src/query/expression/Cargo.toml [hunk 1]
```diff
@@ -38,6 +39,7 @@ geo = { workspace = true }
 geos = { workspace = true }
 geozero = { workspace = true }
 hex = "0.4.3"
+hyper = "0.14"
```
Comment (Xuanwo) [inline src/query/expression/Cargo.toml:42]: Hi, it's better not to depend on hyper "0.14" anymore.

I'm currently working on a upgrade for this.

Comment (everpcpc) [reply src/query/expression/Cargo.toml:42]: It seems arrow-flight depends on tonic 0.11, and tonic 0.11 depends on hyper 0.14. We cannot remove it easily.

Comment (everpcpc) [reply src/query/expression/Cargo.toml:42]: Maybe we could try upgrade arrow-flight later?

Comment (Xuanwo) [reply src/query/expression/Cargo.toml]: Yep, I'm working on this now.

**Review by sundy-li — APPROVED** [review]

**Review by Xuanwo — CHANGES_REQUESTED** [review]
As commented, please don't merge.

**Comment by sundy-li** [comment]
ok, please check that #16696 may also use this client construct function

**Comment by everpcpc** [comment]
> As commented, please don't merge.

I thinks there's no need to block this PR.

1. Hyper 0.14 is already pulled by tonic 0.11, and this PR dose not introduce any new usage on it.
2. After upgrading arrow-rs to 0.53 and tonic 0.12, hyper 0.14 would be gone eventually.
3. DNS issue is a little urgent problem for udf client. If upgrading for arrow-rs has a long way to go, we can't wait for it.

cc @Xuanwo

**Comment by Xuanwo** [comment]
> * After upgrading arrow-rs to 0.53 and tonic 0.12, hyper 0.14 would be gone eventually.

Yes, already working on this. This PR can be based on that PR instead.

PR will land in later one hour.

**Comment by Xuanwo** [comment]
Hi, apologies for blocking you @everpcpc here, but the upcoming PR is quite large and has already taken a full day. If possible, I'd really prefer for this PR to happen afterward.

**Comment by everpcpc** [comment]
PTAL cc @Xuanwo

**Review by Xuanwo — APPROVED** [review]
Thank you @everpcpc for working on this!

**Pull request merged**